File: Solution/Vishal_Seshagiri_task1.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions.{col, avg, _}

object Task1 {
  def main(arg:Array[String]):Unit = {

    val output_path = arg(1)
    val input_path = arg(0)
    val spark = SparkSession.builder.master("local[2]").appName("Assignment 1 Task 1").getOrCreate()
    val ratings_data = spark
      .read
      .format("csv")
      .option("header", "true")
      .option("inferSchema", "true")
      //      .load("../../data/ml-latest-small/ratings.csv")
      .load(input_path)

    var required_data = ratings_data.select("movieId", "rating")

    required_data = required_data.groupBy("movieId").agg(avg("rating").alias("rating_avg"))

    required_data.sort(asc("movieId"))
      .coalesce(1)
      .write
      .format("csv")
      .option("header", "true")
      .save(output_path)

  }
}

File: Solution/Vishal_Seshagiri_task1.py
from pyspark.sql import SparkSession
import sys
import pyspark.sql.functions as func

if __name__=="__main__":

    input_file_path = sys.argv[1]
    output_path = sys.argv[2]
    spark = SparkSession \
        .builder \
        .appName("Assignment 1 Task 1")\
        .getOrCreate()

    ratings_data = spark\
        .read\
        .format("csv")\
        .option("header", "true") \
        .load("{input_file_path}".format(input_file_path=input_file_path),\
        inferSchema=True)

    required_data = ratings_data['movieId', 'rating']

    required_data = required_data.groupBy("movieId").agg(func.avg("rating").alias("rating_avg"))

    required_data \
        .sort("movieId", ascending=True) \
        .coalesce(1) \
        .write \
        .format("csv") \
        .save("{output_path}".format(output_path=output_path), header=True)